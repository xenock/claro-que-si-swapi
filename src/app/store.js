import { configureStore } from '@reduxjs/toolkit'
import starshipsReducer from './slices/starshipsSlice'

export default configureStore({
  reducer: {
    starships: starshipsReducer
  }
})
