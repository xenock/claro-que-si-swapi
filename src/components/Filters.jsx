import React from 'react'
import {
  sortBy,
  setOrderFieldName,
  ascendingOrder,
  toggleOrder,
  searchByTerm,
  syncSelectedFields
} from '../app/slices/starshipsSlice'

import Select from './Select'
import Toggle from './Toggle'
import Searcher from './Searcher'
import FieldsSelector from './FieldsSelector'

import styles from './Filters.module.css'

import { useSelector, useDispatch } from 'react-redux'

const Filters = ({ fields }) => {
  const dispatch = useDispatch()
  const isOn = useSelector(ascendingOrder)

  return (
    <aside>
      <h2>Filters</h2>
      <fieldset className={styles.fieldset}>
        <legend>Find by field with order</legend>
        <Select
          fields={fields}
          onChangeHandler={selectedField => {
            dispatch(setOrderFieldName(selectedField))
            dispatch(sortBy())
          }}
        />
        <Toggle
          isOn={isOn}
          onChangeHandler={() => {
            dispatch(toggleOrder())
            dispatch(sortBy())
          }}
        />
      </fieldset>
      <fieldset className={styles.fieldset}>
        <legend>Select one field at least and find a term on them</legend>
        <Searcher
          onChangeHandler={searchTerm => {
            dispatch(searchByTerm(searchTerm))
          }}
        />
        <FieldsSelector
          fields={fields}
          onChangeHandler={selectedField => {
            dispatch(syncSelectedFields(selectedField))
          }}
        />
      </fieldset>
    </aside>
  )
}

export default Filters
