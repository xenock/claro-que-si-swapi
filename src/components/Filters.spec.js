import React from 'react'
import { render } from '@testing-library/react'
import { Provider } from 'react-redux'

import store from '../app/store'
import Filters from './Filters'

test('renders Filters component', () => {
  const fields = ['name']
  const { getByText, getByPlaceholderText, container } = render(
    <Provider store={store}>
      <Filters fields={fields} />
    </Provider>
  )

  expect(getByText(/Filters/i)).toBeInTheDocument()
  expect(getByPlaceholderText(/Select one field at least/i)).toBeInTheDocument()
  expect(container).toMatchSnapshot()
})
