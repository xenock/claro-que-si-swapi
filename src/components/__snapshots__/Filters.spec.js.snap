// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Filters component 1`] = `
<div>
  <aside>
    <h2>
      Filters
    </h2>
    <div
      class="div"
    >
      <fieldset
        class="fieldset"
      >
        <legend>
          Find by field with order
        </legend>
        <select
          class="select"
          name="select"
        >
          <option
            value="name"
          >
            name
          </option>
        </select>
        <div
          class="block"
        >
          <label
            class="switch"
          >
            <input
              checked=""
              class="input"
              type="checkbox"
            />
            <span
              class="slider"
            />
          </label>
          <p>
             
            ascending
             
          </p>
        </div>
      </fieldset>
      <fieldset
        class="fieldset"
      >
        <legend>
          Select one field at least and find a term on them
        </legend>
        <label
          class="label"
          for="search-input"
        >
          Select one field at least
        </label>
        <input
          class="input"
          disabled=""
          id="search-input"
          name="searcher"
          placeholder="Select one field at least"
          type="text"
        />
        <fieldset
          class="fieldset"
        >
          <div
            class="div"
          >
            <label
              class="label"
              for="name-field"
            >
              name
            </label>
            <input
              class="input"
              id="name-field"
              name="name"
              type="checkbox"
            />
          </div>
        </fieldset>
      </fieldset>
    </div>
  </aside>
</div>
`;
